vyper/venom/passes/literals_codesize.py

```diff
@@ -0,0 +1,47 @@
+from vyper.utils import evm_not
+from vyper.venom.basicblock import IRLiteral
+from vyper.venom.passes.base_pass import IRPass
+
+
+class ReduceLiteralsCodesize(IRPass):
+    def run_pass(self):
+        for bb in self.function.get_basic_blocks():
+            self._process_bb(bb)
+
+    def _process_bb(self, bb):
```

Comment:
For readability, it might be a good thing to add comments to "separate" the three rules the processing applies, or make them separate methods?